src/http.zig

```diff
@@ -1520,40 +1540,93 @@ pub const InternalState = struct {
         return this.received_last_chunk;
     }
 
-    fn decompressBytes(this: *InternalState, buffer: []const u8, body_out_str: *MutableString) !void {
-        log("Decompressing {d} bytes\n", .{buffer.len});
-
+    fn decompressBytes(this: *InternalState, buffer: []const u8, body_out_str: *MutableString, is_final_chunk: bool) !void {
         defer this.compressed_body.reset();
         var gzip_timer: std.time.Timer = undefined;
 
         if (extremely_verbose)
             gzip_timer = std.time.Timer.start() catch @panic("Timer failure");
 
-        try this.decompressor.updateBuffers(this.encoding, buffer, body_out_str);
-        this.decompressor.readAll(this.isDone()) catch |err| {
-            if (this.isDone() or error.ShortRead != err) {
-                Output.prettyErrorln("<r><red>Decompression error: {s}<r>", .{bun.asByteSlice(@errorName(err))});
-                Output.flush();
-                return err;
+        var still_needs_to_decompress = true;
+
+        if (FeatureFlags.isLibdeflateEnabled()) {
+            // Fast-path: use libdeflate
+            if (is_final_chunk and !this.is_libdeflate_fast_path_disabled and this.encoding.canUseLibDeflate() and this.isDone()) libdeflate: {
+                this.is_libdeflate_fast_path_disabled = true;
```

Comment:
why is the fast path disabled here?

Reply:
libdeflate doesn't support streaming, so we can only use it once